src/Avalonia.X11/Clipboard/ClipboardUriListHelper.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Avalonia.Platform.Storage;
+using Avalonia.Platform.Storage.FileIO;
+
+namespace Avalonia.X11.Clipboard;
+
+internal static class ClipboardUriListHelper
+{
+    private static readonly Encoding s_utf8NoBomEncoding = new UTF8Encoding(false);
+
+    public static IStorageItem[] TryReadFileUriList(Stream source)
+    {
+        try
+        {
+            using var reader = new StreamReader(source, s_utf8NoBomEncoding);
```

Comment:
Doesn't feel right to dispose reader _with the source stream_ (as leaveOpen is false by default).
But I double checked - and this method is the only consumer of the stream, so it should be fine.